site/en/tutorials/customization/graph_optimization.ipynb

```diff
@@ -0,0 +1,375 @@
+    "tf.summary.trace_on(graph=True, profiler=True)\n",
+    "\n",
+    "arg = tf.constant(8.9)\n",
+    "print(simple_func(arg))\n",
+    "\n",
+    "# Output traced data as TensorBoard summary data.\n",
+    "with writer.as_default():\n",
+    "    tf.summary.trace_export(\"summary\", step=0, profiler_outdir=\"summary\")\n",
+    "\n",
+    "# Disable tracing data.\n",
+    "tf.summary.trace_off()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, launch TensorBoard to inspect the optimized graph.\n",
+    "\n",
+    "The optimized graph is inspectable via Profile graph.\n",
+    "The first image shows the user defined graph, and the second image shows the optimized graph.\n",
+    "We can notice that mul node and it's input (Const) nodes are missing in the optimized graph.\n",
+    "This indicates that Constant Folding works well against the user defined graph.\n",
+    "\n",
+    "![User Defined Graph](user_defined_graph.png)\n",
+    "\n",
+    "![Optimized Graph](optimized_graph.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Enable/Disable Graph Optimization\n",
+    "\n",
+    "Most of graph optimizations can be enabled/disabled by calling `tf.config.optimizer.set_experimental_options()`.\n",
+    "\n",
+    "Below simple code shows how to enable/disable the optimization.\n",
+    "\n",
+    "* Disable \"Debug Stripper\"\n",
+    "\n",
+    "**Debug Stripper** which is disabled by default, strips the operations used for the debug purpose (Assert, CheckNumerics, Print).\n",
+    "Below code will raise an exception at `tf.Assert`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "\n",
+    "@tf.function\n",
+    "def assert_func():\n",
+    "    a = tf.constant(1.2)\n",
+    "    computation_graph = tf.Assert(tf.less_equal(a, 1.0), [a])   # Will raise an \"InvalidArgumentError\" exception.\n",
+    "    return a\n",
+    "\n",
+    "print(assert_func())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you make Debug Stripper enabled, below code does not raise any exceptions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf\n",
+    "\n",
+    "# Enable \"Debug Stripper\".\n",
+    "tf.config.optimizer.set_experimental_options({'debug_stripper': True})\n",
+    "\n",
+    "@tf.function\n",
+    "def assert_func():\n",
+    "    a = tf.constant(1.2)\n",
+    "    computation_graph = tf.Assert(tf.less_equal(a, 1.0), [a])   # No exceptions are raised.\n",
+    "    return a\n",
+    "\n",
+    "print(assert_func())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* Disable All Graph Optimizations\n",
```

Comment:
This is not getting rendered right, I think

Reply:
Thanks. I fixed to use h3 for emphasizing.